internal/contour/translator_test.go

```diff
@@ -1289,6 +1313,99 @@ func TestTranslatorAddRouteCRD(t *testing.T) {
 	}
 }
 
+func TestTranslatorUpdateStatus(t *testing.T) {
+	tests := []struct {
+		name           string
+		setup          func(*Translator)
+		route          *v1alpha1.Route
+		expectedRoute  *v1alpha1.Route
+		expectedStatus string
+	}{
+		{
+			name: "basic valid",
+			route: &v1alpha1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "simple",
+					Namespace: "default",
+				},
+				Spec: v1alpha1.RouteSpec{
+					Host: "foo.bar",
+					Routes: []v1alpha1.IngressRoute{
+						v1alpha1.IngressRoute{
+							PathPrefix: "/",
+							Upstreams: []v1alpha1.Upstream{
+								v1alpha1.Upstream{
+									ServiceName: "backend",
+									ServicePort: 80,
+								},
+							},
+						},
+					},
+				},
+				Status: v1alpha1.Status{
+					CurrentStatus: "foo",
+				},
+			},
+			expectedRoute: &v1alpha1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "simple",
+					Namespace: "default",
+				},
+				Spec: v1alpha1.RouteSpec{
+					Host: "foo.bar",
+					Routes: []v1alpha1.IngressRoute{
+						v1alpha1.IngressRoute{
+							PathPrefix: "/",
+							Upstreams: []v1alpha1.Upstream{
+								v1alpha1.Upstream{
+									ServiceName: "backend",
+									ServicePort: 80,
+								},
+							},
+						},
+					},
+				},
+				Status: v1alpha1.Status{
+					CurrentStatus: "bar",
```

Comment:
Should this be `Valid`?

Reply:
The intent was that the status should be updated, I took out the status `bar` since it is confusing to the test to make more clear.